pwiz_tools/Skyline/Controls/TargetColumn.cs

```diff
@@ -71,5 +127,284 @@ public override object Clone()
                 return targetCell;
             }
         }
+
+    }
+
+    /// <summary>
+    /// Interface which guarantees automatic management of small molecule details in
+    /// data grids originally designed for peptides
+    /// </summary>
+    public interface ISmallMoleculeColumnsManagementProvider
+    {
+        TargetResolver TargetResolver { get; }
+        SmallMoleculeColumnsManager SmallMoleculeColumnsManager { get; }
     }
+
+    /// <summary>
+    /// Handles updating small molecule detail columns when target is set
+    /// </summary>
+    public class SmallMoleculeColumnsManager
```

Comment:
This class should go in a separate file.

Reply:
Done. Anything else?